File: src/preview_template.py
import os
import sys
import subprocess

# Add the current directory to sys.path to allow imports from lib
# This ensures it works when run from project root as `python src/preview_template.py`
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from lib.pdf_converter import convert_html_to_pdf

def main():
    base_dir = os.path.dirname(os.path.abspath(__file__))
    project_root = os.path.dirname(base_dir)
    
    # 基本テンプレートは src/base に配置
    template_dir = os.path.join(base_dir, 'base')
    input_html = os.path.join(template_dir, 'text.html')
    output_dir = os.path.join(project_root, 'output')
    output_pdf = os.path.join(output_dir, 'preview_template.pdf')

    if not os.path.exists(output_dir):
        os.makedirs(output_dir)

    if not os.path.exists(input_html):
        print(f"Error: Template file not found at {input_html}")
        return

    print(f"Converting {input_html} to PDF...")
    
    # Convert
    # resource_dir is set to template_dir so it can find style.css
    convert_html_to_pdf(input_html, output_pdf, template_dir)

    # Open with default viewer (likely Adobe Reader on Windows)
    if os.path.exists(output_pdf):
        print(f"Opening {output_pdf} with default viewer...")
        try:
            if sys.platform == 'win32':
                os.startfile(output_pdf)
            elif sys.platform == 'darwin':
                subprocess.run(['open', output_pdf], check=True)
            else:
                subprocess.run(['xdg-open', output_pdf], check=True)
        except Exception as e:
            print(f"Failed to open PDF: {e}")
            print("Please open it manually.")

if __name__ == '__main__':
    main()

File: src/preview_template.js
/**
 * 裁判文書の基本テンプレート（src/base/base.html）をプレビュー用のPDFに変換するプログラム。
 * 
 * 使い方:
 *   node src/preview_template.js
 */
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');
const { convertHtmlToPdf } = require('./lib/pdf_converter.js');
const { renderPreTags } = require('./lib/markdown_renderer.js');

async function main() {
    const baseDir = __dirname;
    
    // 基本テンプレートは src/base に配置
    const templateDir = path.join(baseDir, 'base');
    const inputHtml = path.join(templateDir, 'base.html');
    const outputPdf = path.join(templateDir, 'preview.pdf');
    const tempHtml = path.join(templateDir, 'preview.html');

    if (!fs.existsSync(templateDir)) {
        console.error(`Error: Template directory not found at ${templateDir}`);
        return;
    }

    if (!fs.existsSync(inputHtml)) {
        console.error(`Error: Template file not found at ${inputHtml}`);
        return;
    }

    console.log(`Rendering Markdown and converting ${inputHtml} to PDF...`);
    
    // Markdownの事前レンダリング
    try {
        const htmlContent = fs.readFileSync(inputHtml, 'utf-8');
        const renderedContent = renderPreTags(htmlContent, templateDir);
        
        // 一時ファイルに保存
        fs.writeFileSync(tempHtml, renderedContent, 'utf-8');
        
        // Convert
        // resourceDir is set to templateDir so it can find style.css
        await convertHtmlToPdf(tempHtml, outputPdf, templateDir);

        console.log(`一時ファイルを保存しました: ${tempHtml}`);
    } catch (err) {
        console.error(`Error during rendering/conversion: ${err}`);
        return;
    }

    // Open with default viewer
    if (fs.existsSync(outputPdf)) {
        console.log(`Opening ${outputPdf} with default viewer...`);
        const platform = process.platform;
        let command;
        
        if (platform === 'win32') {
            command = `start "" "${outputPdf}"`;
        } else if (platform === 'darwin') {
            command = `open "${outputPdf}"`;
        } else {
            command = `xdg-open "${outputPdf}"`;
        }

        exec(command, (err) => {
            if (err) {
                console.error(`Failed to open PDF: ${err}`);
                console.log("Please open it manually.");
            }
        });
    }
}

main();
